platform/ios/src/UIColor+MGLAdditions.mm

```diff
@@ -17,7 +18,13 @@ @implementation UIColor (MGLAdditions)
 
 + (UIColor *)mgl_colorWithColor:(mbgl::Color)color
 {
-    return [UIColor colorWithRed:color.r green:color.g blue:color.b alpha:color.a];
+    // Since CIColor expects color values to be non-multiplied, we transform them here
+    // to non-premultiplied values
+    float red = static_cast<float>((color.r / color.a));
+    float green = static_cast<float>((color.g / color.a));
+    float blue = static_cast<float>((color.b / color.a));
```

Comment:
If the alpha is zero, these lines divide by zero. I think this is why the tests are failing:

```
  testProperties, ((layer.fillOutlineColor) equal to (defaultExpression)) failed: ("UIExtendedSRGBColorSpace nan nan nan 0") is not equal to ("UIExtendedSRGBColorSpace nan nan nan 0") - fillOutlineColor should return the default value after being unset.
```

Reply:
Let's return black with an alpha of 0 here here (`0,0,0,0`). 

Though @alexshalamov asked a good question - do we need to preserve the original color information? If so, we're going to need something beyond `mbgl::Color`.

Can you think of situations where this will crop up?

Reply:
@1ec5 @julianrex @alexshalamov thank you all for the comments, we already discussed a fix yesterday after Nadia made this pr, and goes along with what Alex wrote here https://github.com/mapbox/mapbox-gl-native/pull/14406/files#r274946020